From: Tariq Toukan <tariqt@mellanox.com>
Date: Tue, 10 Oct 2017 16:51:44 +0300
Subject: net/mlx5e: Add error print in ETS init
Patch-mainline: v4.15-rc9
Git-commit: 97c8c3aa48ca8eb85d1806e08f882f90d78b1856
References: bsc#1046303 FATE#322944

ETS initialization might fail, add a print to indicate
such failures.

Fixes: 08fb1dacdd76 ("net/mlx5e: Support DCBNL IEEE ETS")
Signed-off-by: Tariq Toukan <tariqt@mellanox.com>
Signed-off-by: Saeed Mahameed <saeedm@mellanox.com>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/net/ethernet/mellanox/mlx5/core/en_dcbnl.c |    8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

--- a/drivers/net/ethernet/mellanox/mlx5/core/en_dcbnl.c
+++ b/drivers/net/ethernet/mellanox/mlx5/core/en_dcbnl.c
@@ -784,8 +784,9 @@ static void mlx5e_dcbnl_query_dcbx_mode(
 
 static void mlx5e_ets_init(struct mlx5e_priv *priv)
 {
-	int i;
 	struct ieee_ets ets;
+	int err;
+	int i;
 
 	if (!MLX5_CAP_GEN(priv->mdev, ets))
 		return;
@@ -802,7 +803,10 @@ static void mlx5e_ets_init(struct mlx5e_
 	ets.prio_tc[0] = 1;
 	ets.prio_tc[1] = 0;
 
-	mlx5e_dcbnl_ieee_setets_core(priv, &ets);
+	err = mlx5e_dcbnl_ieee_setets_core(priv, &ets);
+	if (err)
+		netdev_err(priv->netdev,
+			   "%s, Failed to init ETS: %d\n", __func__, err);
 }
 
 void mlx5e_dcbnl_initialize(struct mlx5e_priv *priv)
